superset/datasets/api.py

```diff
@@ -18,11 +18,16 @@
 import logging
 from datetime import datetime
 from io import BytesIO
-from typing import Any
+from typing import Any, Optional
```

Comment:
Hint: you can do `from __future__ import annotations` and then use `a | None` instead of `Optional[a]`.